agent/agent.go

```diff
@@ -1485,15 +1485,25 @@ func writeFileAtomic(path string, contents []byte) error {
 		return err
 	}
 	if _, err := fh.Write(contents); err != nil {
+		fh.Close()
+		os.Remove(tempPath)
 		return err
 	}
 	if err := fh.Sync(); err != nil {
+		fh.Close()
+		os.Remove(tempPath)
 		return err
 	}
 	if err := fh.Close(); err != nil {
+		fh.Close()
```

Comment:
@preetapan I realized that this close is redundant, too - removed it in https://github.com/hashicorp/consul/commit/c413a9161e496a2b02e28364f6d9ae9c4b5fef3a